File: main2.java
import java.util.Scanner;

public class Main {
    
    public static String punkt(String[][] v) {
        int n = v.length;
        
        String start = "";
        for (int i = 0; i < n; i++) {
            boolean ok = true;
            for (int j = 0; j < n; j++) {
                if (v[j][1].equals(v[i][0])) {
                    ok = false;
                    break;
                }
            }
            if (ok) {
                start = v[i][0];
                break;
            }
        }
        
        if (start.equals("")) {
            return "Ошибка";
        }
        
        String cur = start;
        for (int step = 0; step < n; step++) {
            for (int i = 0; i < n; i++) {
                if (v[i][0].equals(cur)) {
                    cur = v[i][1];
                    break;
                }
            }
        }
        
        if (cur.equals(start)) {
            return "Ошибка";
        }
        return cur;
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Введите количество пар: ");
        int n = scanner.nextInt();
        scanner.nextLine(); // очистка буфера
        
        String[][] pairs = new String[n][2];
        
        System.out.println("Введите пары (начало и конец):");
        for (int i = 0; i < n; i++) {
            pairs[i][0] = scanner.next();
            pairs[i][1] = scanner.next();
        }
        
        System.out.println(punkt(pairs));
        
        scanner.close();
    }
}

File: main2.kt
import java.util.Scanner

fun punkt(v: Array<Array<String>>): String {
    val n = v.size
    
    var start = ""
    for (i in 0 until n) {
        var ok = true
        for (j in 0 until n) {
            if (v[j][1] == v[i][0]) {
                ok = false
                break
            }
        }
        if (ok) {
            start = v[i][0]
            break
        }
    }
    
    if (start == "") {
        return "Ошибка"
    }
    
    var cur = start
    for (step in 0 until n) {
        for (i in 0 until n) {
            if (v[i][0] == cur) {
                cur = v[i][1]
                break
            }
        }
    }
    
    return if (cur == start) "Ошибка" else cur
}

fun main() {
    val scanner = Scanner(System.`in`)
    
    print("Введите количество пар: ")
    val n = scanner.nextInt()
    
    val pairs = Array(n) { arrayOfNulls<String>(2) }
    
    println("Введите пары (начало и конец):")
    for (i in 0 until n) {
        pairs[i][0] = scanner.next()
        pairs[i][1] = scanner.next()
    }
    
    println(punkt(pairs.map { arrayOf(it[0]!!, it[1]!!) }.toTypedArray()))
}